arbeitszeit/use_cases/query_plans.py

```diff
@@ -64,11 +66,14 @@ class QueryPlans:
 
     def __call__(self, request: QueryPlansRequest) -> PlanQueryResponse:
         now = self.datetime_service.now()
-        plans = (
-            self.database_gateway.get_plans()
-            .that_will_expire_after(now)
-            .that_were_activated_before(now)
-        )
+        if request.include_expired_plans:
```

Comment:
The `_plan_to_response_model` method takes as first argument only these plans that have already been filtered beforehand. So including or excluding the expired plans has to happen before calling this method.
But your comment made me aware that this can be written shorter and better readable:

```
plans = self.database_gateway.get_plans().that_were_activated_before(now)
if not request.include_expired_plans:
    # exclude expired plans
    plans = plans.that_will_expire_after(now)
```

Reply:
Looking at the code again also made me aware that the `total_results` response attribute, which is currently only used for pagination, is probably calculated wrongly. The line `total_results = len(plans)` should come after all the filtering took place. I'll investigate it and fix it or write a ticket if this is really a bug.

Reply:
This is totally fine for me. What I meant was just that you are calculating the is_expired property anyway in _plan_to_response_model, so no matter which plans will be passed (expired or not), the is_expired property will always be set correctly.